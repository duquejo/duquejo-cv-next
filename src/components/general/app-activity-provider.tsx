'use client';

import { ReactNode, useState } from 'react';
import {
  Sheet,
  SheetContent,
  SheetDescription,
  SheetFooter,
  SheetHeader,
  SheetTitle,
} from '@/components/ui/sheet';
import Link from 'next/link';
import { EventSkeleton } from '@/components/events/event-skeleton';
import useSWR from 'swr';
import { getEvents } from '@/actions/github';
import { Event } from '@/interfaces';
import { EventCard } from '@/components/events/event-card';
import { useTranslations } from 'next-intl';

interface Props {
  children: ReactNode;
  defaultOpen?: boolean;
}

export const AppActivityProvider = ({ children, defaultOpen = false }: Props) => {
  const t = useTranslations('Sidebar.complementary.activity.dialog');
  const [loadEvents, setLoadEvents] = useState(false);

  const { data, isLoading } = useSWR<Event[]>(loadEvents ? 'events' : null, getEvents);

  const onToggleChange = async (open: boolean) => {
    if (open) {
      return setLoadEvents(true);
    }
    setLoadEvents(false);
  };

  return (
    <Sheet key="activity" onOpenChange={onToggleChange} defaultOpen={defaultOpen}>
      {children}
      <SheetContent role="dialog" className="flex flex-col justify-between w-10/12">
        <SheetHeader>
          <SheetTitle>{t('title')}</SheetTitle>
          <SheetDescription>{t('description')}</SheetDescription>
        </SheetHeader>
        <div className="flex-1 flex flex-col gap-2 overflow-y-auto">
          {isLoading ? (
            <EventSkeleton rounds={2} />
          ) : (
            <div role="list" className="flex flex-col gap-2 overflow-y-auto list-none">
              {!data || data.length === 0 ? (
                <p data-testid="empty" className="text-muted-foreground text-sm">
                  {t('empty')}
                </p>
              ) : (
                data.map((event) => <EventCard key={event.id} {...event} />)
              )}
            </div>
          )}
        </div>
        <SheetFooter>
          <Link
            href="https://github.com/duquejo"
            target="_blank"
            className="text-xs text-muted-foreground"
          >
            {t('button')}
          </Link>
        </SheetFooter>
      </SheetContent>
    </Sheet>
  );
};
