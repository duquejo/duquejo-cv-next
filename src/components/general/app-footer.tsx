'use client';

import Link from 'next/link';
import { Social } from '@/interfaces';
import { Github, Linkedin } from 'lucide-react';

export const SOCIAL_DATA: Social[] = [
  {
    name: 'GitHub',
    link: 'https://github.com/duquejo',
    icon: <Github />,
  },
  {
    name: 'LinkedIn',
    link: 'https://www.linkedin.com/in/duquejo/',
    icon: <Linkedin />,
  },
];

export const AppFooter = () => {
  return (
    <footer className="border-t flex p-3 items-center justify-center gap-x-8">
      {SOCIAL_DATA.map((social) => (
        <Link
          href={social.link}
          key={social.name}
          target="_blank"
          title={social.name}
          aria-label={social.name}
        >
          {social.icon}
        </Link>
      ))}
      <span className="font-semibold text-sm">© {new Date().getFullYear()} José Duque</span>
    </footer>
  );
};
