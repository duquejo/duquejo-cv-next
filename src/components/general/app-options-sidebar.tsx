import { ThemeToggle } from '@/components/general/theme-toggle';
import { SidebarTrigger } from '@/components/ui/sidebar';

export const AppOptionsSidebar = () => {
  return (
    <nav className="lg:w-14 lg:max-w-14 lg:h-screen lg:relative w-auto fixed right-0 lg:mr-1">
      <div className="flex flex-row gap-2 bg-sidebar rounded-lg p-2 m-2 lg:mx-0 lg:mb-2 lg:flex-col lg:fixed lg:bottom-0 lg:border lg:p-1">
        <SidebarTrigger
          variant="outline"
          className="h-8 w-8 lg:h-10 lg:w-10"
          title="Toggle sidebar"
        />
        <ThemeToggle
          variant="outline"
          className="h-8 w-8 lg:h-10 lg:w-10"
          title="Change color theme"
        />
      </div>
    </nav>
  );
};
