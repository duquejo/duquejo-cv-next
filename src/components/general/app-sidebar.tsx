import {
  Sidebar,
  SidebarContent,
  SidebarGroup,
  SidebarGroupContent,
  SidebarGroupLabel,
  SidebarHeader,
  SidebarMenu,
  SidebarSeparator,
} from '@/components/ui/sidebar';
import { MenuItem } from '@/interfaces';
import { Calendar, Gamepad, Inbox, Music, Pencil, Projector, Send } from 'lucide-react';
import { MenuAvatar } from '@/components/resume/menu-avatar/MenuAvatar';
import { MenuItems } from '@/components/resume/menu-items/MenuItems';

// Menu items.
export const professionalItems: MenuItem[] = [
  {
    title: 'Resume√©',
    url: '/',
    icon: <Inbox />,
  },
  {
    title: 'Services',
    url: '/career/services',
    icon: <Calendar />,
  },
  {
    title: 'Latest projects',
    url: '/career/projects',
    icon: <Projector />,
  },
];

export const lastItems: MenuItem[] = [
  {
    title: 'My blog',
    url: '/blog',
    icon: <Pencil />,
  },
  {
    title: 'Contact',
    url: '/contact',
    icon: <Send />,
  },
];

export const hobbiesItems: MenuItem[] = [
  {
    title: 'Music production',
    url: '/hobbies/music-production',
    icon: <Music />,
  },
  {
    title: 'Game development',
    url: '/hobbies/game-development',
    icon: <Gamepad />,
  },
];

export const AppSidebar = () => {
  return (
    <Sidebar collapsible="icon" variant="floating" role="navigation" tabIndex={0}>
      <SidebarHeader>
        <div className="flex flex-col place-items-center gap-y-2">
          <MenuAvatar />
        </div>
      </SidebarHeader>
      <SidebarContent className="justify-between">
        <div>
          <SidebarGroup>
            <SidebarGroupLabel>Professional career</SidebarGroupLabel>
            <SidebarGroupContent>
              <SidebarMenu>
                <MenuItems items={professionalItems} />
              </SidebarMenu>
            </SidebarGroupContent>
          </SidebarGroup>
          <SidebarGroup>
            <SidebarGroupLabel>Hobbies</SidebarGroupLabel>
            <SidebarGroupContent>
              <SidebarMenu>
                <MenuItems items={hobbiesItems} />
              </SidebarMenu>
            </SidebarGroupContent>
          </SidebarGroup>
        </div>
        <div>
          <SidebarSeparator />
          <SidebarGroup>
            <SidebarGroupContent>
              <SidebarMenu>
                <MenuItems size="sm" items={lastItems} />
              </SidebarMenu>
            </SidebarGroupContent>
          </SidebarGroup>
        </div>
      </SidebarContent>
    </Sidebar>
  );
};
