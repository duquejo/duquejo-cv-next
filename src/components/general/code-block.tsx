import { Light as SyntaxHighlighter } from 'react-syntax-highlighter';
import ts from 'react-syntax-highlighter/dist/esm/languages/hljs/typescript';
import atomDark from 'react-syntax-highlighter/dist/cjs/styles/hljs/atom-one-dark-reasonable';

SyntaxHighlighter.registerLanguage('javascript', ts);

interface CodeBlockProps {
  code: string;
}

export const CodeBlock = ({ code }: CodeBlockProps) => {
  return (
    <SyntaxHighlighter
      style={atomDark}
      showLineNumbers
      customStyle={{ margin: 0, borderRadius: 0, fontSize: 14 }}
    >
      {code}
    </SyntaxHighlighter>
  );
};
