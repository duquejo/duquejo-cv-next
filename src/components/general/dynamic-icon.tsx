import type { ComponentType } from 'react';
import dynamic from 'next/dynamic';
import { LoaderCircle } from 'lucide-react';

interface Props {
  iconName: string;
  [key: string]: any; // eslint-disable-line @typescript-eslint/no-explicit-any
}
type LucideIconsModule = Record<string, ComponentType<unknown>>;

export const DynamicIcon = ({ iconName, ...props }: Props) => {
  const Icon = dynamic(
    () =>
      import('lucide-react').then((module) => {
        const iconsModule = module as unknown as LucideIconsModule;
        const component = iconsModule[iconName];
        if (!component) {
          throw new Error(`Icon ${iconName} not found`);
        }
        return component;
      }),
    {
      loading: () => (
        <LoaderCircle size="70" strokeWidth={1} className="text-secondary-foreground" />
      ),
    },
  );

  return <Icon {...props} />;
};
