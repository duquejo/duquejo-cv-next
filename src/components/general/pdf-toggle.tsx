'use client';

import { Button, type ButtonProps } from '@/components/ui/button';
import { Popover, PopoverContent, PopoverTrigger } from '@/components/ui/popover';
import { ImageDown } from 'lucide-react';
import { cn } from '@/lib';
import { PdfForm } from '@/components/pdf/pdf-form';
import { useState } from 'react';

interface Props {
  className?: string;
  buttonTitle: string;
  formDescription: string;
  formTitle: string;
  formButtonText: string;
  formButtonTextLoading: string;
  variant?: ButtonProps['variant'];
}

export const PdfGeneratorToggle = ({
  formDescription,
  formButtonText,
  formTitle,
  buttonTitle,
  formButtonTextLoading,
  className,
  variant,
}: Props) => {
  const [isPopoverOpen, setIsPopoverOpen] = useState(false);

  return (
    <Popover key="pdf" open={isPopoverOpen}>
      <PopoverTrigger asChild>
        <Button
          variant={variant}
          size="icon"
          className={cn('h-10 w-10', className)}
          onClick={() => setIsPopoverOpen(true)}
        >
          <ImageDown className="h-8 w-8 lg:h-10 lg:w-10" aria-label={buttonTitle} />
          <span className="sr-only">{buttonTitle}</span>
        </Button>
      </PopoverTrigger>
      <PopoverContent
        role="dialog"
        aria-label={formTitle}
        onInteractOutside={() => setIsPopoverOpen(false)}
        onEscapeKeyDown={() => setIsPopoverOpen(false)}
      >
        <h2 id="form-title" className="font-medium leading-none mb-2 text-sm">
          {formTitle}
        </h2>
        <p id="form-description" className="text-muted-foreground text-xs">
          {formDescription}
        </p>
        <PdfForm
          onSubmitFinish={() => setIsPopoverOpen(false)}
          button={formButtonText}
          buttonLoading={formButtonTextLoading}
        />
      </PopoverContent>
    </Popover>
  );
};
