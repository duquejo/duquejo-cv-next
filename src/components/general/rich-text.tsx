// @flow
import * as React from 'react';
import { ReactNode } from 'react';

type Tag = 'b' | 'br';

interface Props {
  children(tags: Record<Tag, (chunks: ReactNode) => ReactNode>): ReactNode;
}
export const RichText = ({ children }: Props) => {
  return children({
    b: (chunks: ReactNode) => <b>{chunks}</b>,
    br: () => <br />,
  });
};
