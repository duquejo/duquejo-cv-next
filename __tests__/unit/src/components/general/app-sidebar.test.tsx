import { render, screen } from '@testing-library/react';
import { AppSidebar } from '@/components/general/app-sidebar';
import { SidebarProvider } from '@/components/ui/sidebar';

const professionalItems = [
  { title: 'Home', url: '/home' },
  { title: 'About', url: '/about' },
];

const hobbiesItems = [
  { title: 'Reading', url: '/reading' },
  { title: 'Gaming', url: '/gaming' },
];

const footerItems = [{ title: 'Contact', url: '/contact' }];

vi.mock('@/i18n/routing', () => ({
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  Link: ({ children, href, onClick, ...props }: any) => (
    <a href={href} onClick={onClick} {...props}>
      {children}
    </a>
  ),
  usePathname: vi.fn(() => '/home'),
}));

vi.mock('next-intl', () => ({
  useTranslations: () => {
    const translations: Record<string, unknown> = {
      role: 'User Role',
      'professional.title': 'Professional Links',
      'hobbies.title': 'Hobbies',
      'professional.links': professionalItems,
      'hobbies.links': hobbiesItems,
      'footer.links': footerItems,
    };

    const t = (key: string) => translations[key];

    t.raw = (key: string) => translations[key];

    return t;
  },
}));

describe('<AppSidebar /> tests', () => {
  beforeEach(() => {
    vi.clearAllMocks();
    window.matchMedia = vi.fn().mockImplementation((query) => ({
      matches: query === '(min-width: 768px)',
      media: query,
      onchange: null,
      addListener: vi.fn(),
      removeListener: vi.fn(),
      addEventListener: vi.fn(),
      removeEventListener: vi.fn(),
      dispatchEvent: vi.fn(),
    }));
  });

  it('Should match the snapshot', () => {
    const { container } = render(
      <SidebarProvider>
        <AppSidebar />
      </SidebarProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render the required sections 'Avatar', 'Professional career', 'Hobbies' and misc.", () => {
    render(
      <SidebarProvider>
        <AppSidebar />
      </SidebarProvider>,
    );

    expect(screen.getByText('Professional Links')).toBeInTheDocument();
    expect(screen.getByText('Hobbies')).toBeInTheDocument();
    expect(screen.getByRole('img')).toBeInTheDocument();

    professionalItems.forEach(({ title, url }) => {
      const screenItem = screen.getByRole('link', { name: title });
      expect(screenItem).toBeInTheDocument();
      expect(screenItem).toHaveTextContent(title);
      expect(screenItem).toHaveAttribute('href', url);
    });

    hobbiesItems.forEach(({ title, url }) => {
      const screenItem = screen.getByRole('link', { name: title });
      expect(screenItem).toBeInTheDocument();
      expect(screenItem).toHaveTextContent(title);
      expect(screenItem).toHaveAttribute('href', url);
    });

    footerItems.forEach(({ title, url }) => {
      const screenItem = screen.getByRole('link', { name: title });
      expect(screenItem).toBeInTheDocument();
      expect(screenItem).toHaveTextContent(title);
      expect(screenItem).toHaveAttribute('href', url);
    });
  });
});
