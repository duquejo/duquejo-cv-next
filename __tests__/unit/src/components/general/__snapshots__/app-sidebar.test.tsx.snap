// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<AppSidebar /> tests > Should match the snapshot 1`] = `
<div>
  <div
    class="group/sidebar-wrapper flex min-h-svh w-full has-[[data-variant=inset]]:bg-sidebar"
    style="--sidebar-width: 18rem; --sidebar-width-icon: 3rem;"
  >
    <div
      class="group peer hidden md:block text-sidebar-foreground"
      data-collapsible=""
      data-side="left"
      data-state="expanded"
      data-variant="floating"
    >
      <div
        class="duration-200 relative h-svh w-[--sidebar-width] bg-transparent transition-[width] ease-linear group-data-[collapsible=offcanvas]:w-0 group-data-[side=right]:rotate-180 group-data-[collapsible=icon]:w-[calc(var(--sidebar-width-icon)_+_theme(spacing.4))]"
      />
      <div
        class="duration-200 fixed inset-y-0 z-10 hidden h-svh w-[--sidebar-width] transition-[left,right,width] ease-linear md:flex left-0 group-data-[collapsible=offcanvas]:left-[calc(var(--sidebar-width)*-1)] p-2 group-data-[collapsible=icon]:w-[calc(var(--sidebar-width-icon)_+_theme(spacing.4)_+2px)]"
        role="navigation"
        tabindex="0"
      >
        <div
          class="flex h-full w-full flex-col bg-sidebar group-data-[variant=floating]:rounded-lg group-data-[variant=floating]:border group-data-[variant=floating]:border-sidebar-border group-data-[variant=floating]:shadow"
          data-sidebar="sidebar"
        >
          <div
            class="flex flex-col gap-2 p-2"
            data-sidebar="header"
          >
            <div
              class="flex flex-col place-items-center gap-y-2"
            >
              <picture
                class="mt-10 peer-data-[active=true]/menu-button:w-6 max-w-32 lg:w-auto"
              >
                <img
                  alt="José Miguel Duque"
                  class="object-cover rounded"
                  data-nimg="1"
                  decoding="async"
                  height="146"
                  src="/_next/image?url=%2Fstatic%2Fimg%2Favatar-small.webp&w=384&q=75"
                  srcset="/_next/image?url=%2Fstatic%2Fimg%2Favatar-small.webp&w=256&q=75 1x, /_next/image?url=%2Fstatic%2Fimg%2Favatar-small.webp&w=384&q=75 2x"
                  style="color: transparent;"
                  width="160"
                />
              </picture>
              <div
                class="text-center"
              >
                <span
                  class="relative w-[max-content] font-mono before:absolute before:inset-0 before:animate-typewriter before:opacity-80 after:absolute after:inset-0 after:w-[0.125em] after:animate-caret after:bg-primary  after:ml-0.5 font-light text-base text-primary"
                >
                  @duquejo
                </span>
                <div
                  class="text-xl font-bold"
                >
                  José Duque
                </div>
                <div
                  class="font-light text-sm"
                >
                  User Role
                </div>
              </div>
            </div>
          </div>
          <div
            class="flex min-h-0 flex-1 flex-col gap-2 overflow-auto group-data-[collapsible=icon]:overflow-hidden justify-between"
            data-sidebar="content"
          >
            <div>
              <div
                class="relative flex w-full min-w-0 flex-col p-2"
                data-sidebar="group"
              >
                <div
                  class="duration-200 flex h-8 shrink-0 items-center rounded-md px-2 text-xs font-medium text-sidebar-foreground/70 outline-none ring-sidebar-ring transition-[margin,opa] ease-linear focus-visible:ring-2 [&>svg]:size-4 [&>svg]:shrink-0 group-data-[collapsible=icon]:-mt-8 group-data-[collapsible=icon]:opacity-0"
                  data-sidebar="group-label"
                >
                  Professional Links
                </div>
                <div
                  class="w-full text-sm"
                  data-sidebar="group-content"
                >
                  <ul
                    class="flex w-full min-w-0 flex-col gap-1"
                    data-sidebar="menu"
                  >
                    <ul
                      class="flex w-full min-w-0 flex-col gap-1"
                      data-sidebar="menu"
                    >
                      <li
                        class="group/menu-item relative"
                        data-sidebar="menu-item"
                      >
                        <a
                          class="peer/menu-button flex w-full items-center gap-2 overflow-hidden rounded-md p-2 text-left outline-none ring-sidebar-ring transition-[width,height,padding] active:bg-sidebar-accent active:text-sidebar-accent-foreground disabled:pointer-events-none disabled:opacity-50 group-has-[[data-sidebar=menu-action]]/menu-item:pr-8 aria-disabled:pointer-events-none aria-disabled:opacity-50 data-[active=true]:bg-sidebar-accent data-[active=true]:font-medium data-[active=true]:text-sidebar-accent-foreground data-[state=open]:hover:bg-sidebar-accent data-[state=open]:hover:text-sidebar-accent-foreground group-data-[collapsible=icon]:!size-8 group-data-[collapsible=icon]:!p-2 [&>span:last-child]:truncate [&>svg]:size-4 [&>svg]:shrink-0 hover:bg-sidebar-accent hover:text-sidebar-accent-foreground h-8 text-sm focus-visible:ring-primary focus-visible:ring-1"
                          data-active="true"
                          data-sidebar="menu-button"
                          data-size="default"
                          href="/home"
                        >
                          <span>
                            Home
                          </span>
                        </a>
                      </li>
                      <li
                        class="group/menu-item relative"
                        data-sidebar="menu-item"
                      >
                        <a
                          class="peer/menu-button flex w-full items-center gap-2 overflow-hidden rounded-md p-2 text-left outline-none ring-sidebar-ring transition-[width,height,padding] active:bg-sidebar-accent active:text-sidebar-accent-foreground disabled:pointer-events-none disabled:opacity-50 group-has-[[data-sidebar=menu-action]]/menu-item:pr-8 aria-disabled:pointer-events-none aria-disabled:opacity-50 data-[active=true]:bg-sidebar-accent data-[active=true]:font-medium data-[active=true]:text-sidebar-accent-foreground data-[state=open]:hover:bg-sidebar-accent data-[state=open]:hover:text-sidebar-accent-foreground group-data-[collapsible=icon]:!size-8 group-data-[collapsible=icon]:!p-2 [&>span:last-child]:truncate [&>svg]:size-4 [&>svg]:shrink-0 hover:bg-sidebar-accent hover:text-sidebar-accent-foreground h-8 text-sm focus-visible:ring-primary focus-visible:ring-1"
                          data-active="false"
                          data-sidebar="menu-button"
                          data-size="default"
                          href="/about"
                        >
                          <span>
                            About
                          </span>
                        </a>
                      </li>
                    </ul>
                  </ul>
                </div>
              </div>
              <div
                class="relative flex w-full min-w-0 flex-col p-2"
                data-sidebar="group"
              >
                <div
                  class="duration-200 flex h-8 shrink-0 items-center rounded-md px-2 text-xs font-medium text-sidebar-foreground/70 outline-none ring-sidebar-ring transition-[margin,opa] ease-linear focus-visible:ring-2 [&>svg]:size-4 [&>svg]:shrink-0 group-data-[collapsible=icon]:-mt-8 group-data-[collapsible=icon]:opacity-0"
                  data-sidebar="group-label"
                >
                  Hobbies
                </div>
                <div
                  class="w-full text-sm"
                  data-sidebar="group-content"
                >
                  <ul
                    class="flex w-full min-w-0 flex-col gap-1"
                    data-sidebar="menu"
                  >
                    <ul
                      class="flex w-full min-w-0 flex-col gap-1"
                      data-sidebar="menu"
                    >
                      <li
                        class="group/menu-item relative"
                        data-sidebar="menu-item"
                      >
                        <a
                          class="peer/menu-button flex w-full items-center gap-2 overflow-hidden rounded-md p-2 text-left outline-none ring-sidebar-ring transition-[width,height,padding] active:bg-sidebar-accent active:text-sidebar-accent-foreground disabled:pointer-events-none disabled:opacity-50 group-has-[[data-sidebar=menu-action]]/menu-item:pr-8 aria-disabled:pointer-events-none aria-disabled:opacity-50 data-[active=true]:bg-sidebar-accent data-[active=true]:font-medium data-[active=true]:text-sidebar-accent-foreground data-[state=open]:hover:bg-sidebar-accent data-[state=open]:hover:text-sidebar-accent-foreground group-data-[collapsible=icon]:!size-8 group-data-[collapsible=icon]:!p-2 [&>span:last-child]:truncate [&>svg]:size-4 [&>svg]:shrink-0 hover:bg-sidebar-accent hover:text-sidebar-accent-foreground h-8 text-sm focus-visible:ring-primary focus-visible:ring-1"
                          data-active="false"
                          data-sidebar="menu-button"
                          data-size="default"
                          href="/reading"
                        >
                          <span>
                            Reading
                          </span>
                        </a>
                      </li>
                      <li
                        class="group/menu-item relative"
                        data-sidebar="menu-item"
                      >
                        <a
                          class="peer/menu-button flex w-full items-center gap-2 overflow-hidden rounded-md p-2 text-left outline-none ring-sidebar-ring transition-[width,height,padding] active:bg-sidebar-accent active:text-sidebar-accent-foreground disabled:pointer-events-none disabled:opacity-50 group-has-[[data-sidebar=menu-action]]/menu-item:pr-8 aria-disabled:pointer-events-none aria-disabled:opacity-50 data-[active=true]:bg-sidebar-accent data-[active=true]:font-medium data-[active=true]:text-sidebar-accent-foreground data-[state=open]:hover:bg-sidebar-accent data-[state=open]:hover:text-sidebar-accent-foreground group-data-[collapsible=icon]:!size-8 group-data-[collapsible=icon]:!p-2 [&>span:last-child]:truncate [&>svg]:size-4 [&>svg]:shrink-0 hover:bg-sidebar-accent hover:text-sidebar-accent-foreground h-8 text-sm focus-visible:ring-primary focus-visible:ring-1"
                          data-active="false"
                          data-sidebar="menu-button"
                          data-size="default"
                          href="/gaming"
                        >
                          <span>
                            Gaming
                          </span>
                        </a>
                      </li>
                    </ul>
                  </ul>
                </div>
              </div>
            </div>
            <div>
              <div
                class="shrink-0 h-[1px] mx-2 w-auto bg-sidebar-border"
                data-orientation="horizontal"
                data-sidebar="separator"
                role="none"
              />
              <div
                class="relative flex w-full min-w-0 flex-col p-2"
                data-sidebar="group"
              >
                <div
                  class="w-full text-sm"
                  data-sidebar="group-content"
                >
                  <ul
                    class="flex w-full min-w-0 flex-col gap-1"
                    data-sidebar="menu"
                  >
                    <ul
                      class="flex w-full min-w-0 flex-col gap-1"
                      data-sidebar="menu"
                    >
                      <li
                        class="group/menu-item relative"
                        data-sidebar="menu-item"
                      >
                        <a
                          class="peer/menu-button flex w-full items-center gap-2 overflow-hidden rounded-md p-2 text-left outline-none ring-sidebar-ring transition-[width,height,padding] active:bg-sidebar-accent active:text-sidebar-accent-foreground disabled:pointer-events-none disabled:opacity-50 group-has-[[data-sidebar=menu-action]]/menu-item:pr-8 aria-disabled:pointer-events-none aria-disabled:opacity-50 data-[active=true]:bg-sidebar-accent data-[active=true]:font-medium data-[active=true]:text-sidebar-accent-foreground data-[state=open]:hover:bg-sidebar-accent data-[state=open]:hover:text-sidebar-accent-foreground group-data-[collapsible=icon]:!size-8 group-data-[collapsible=icon]:!p-2 [&>span:last-child]:truncate [&>svg]:size-4 [&>svg]:shrink-0 hover:bg-sidebar-accent hover:text-sidebar-accent-foreground h-7 text-xs focus-visible:ring-primary focus-visible:ring-1"
                          data-active="false"
                          data-sidebar="menu-button"
                          data-size="sm"
                          href="/contact"
                        >
                          <span>
                            Contact
                          </span>
                        </a>
                      </li>
                    </ul>
                  </ul>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
